server/game/effects.js

```diff
@@ -114,13 +114,18 @@ const Effects = {
                 let challenge = context.game.currentChallenge;
                 if(card.canParticipateInChallenge() && !challenge.isAttacking(card)) {
                     challenge.addAttacker(card);
-                }
+                    card.consideredToBeAttackingWasApplied = true;
+                    context.game.once('afterChallenge', () => {
```

Comment:
I think this would actually work, but there are some bad interactions from the effect engine. Since the effect has a `condition`, I think the engine will continually `unapply` and then re-`apply` the effect. So this event handler will be created more than once. Which in terms of what it's doing is fine - worst case the flag is set to `false` multiple times. It could be bad in terms of introducing an unintended number of event listener - I'm not sure whether there are implications in Node for that.

The alternate solution is to record which characters were declared on the `Challenge` class itself, and check that instead of adding this flag. Even doing that, though, there are edge cases - declare Azor Stannis and another R'hllor character, remove Azor Stannis from the challenge. Even though he was declared, since he was removed, Azor Ahai should make him participate again.